File: arrays/find_peak_element_brute_force.rb
def find_peak_element(nums)
  return 0 if nums.size == 1

  nums.each_with_index do |num, i|
    return i if i == 0 && num > nums[i + 1]
    return i if i > 0 && i < nums.size - 1 && num > nums

    return i if i > 0 && i < nums.size - 1 && num > nums
  end

  0
end

File: arrays/find_peak_element_brute_force.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:def find_peak_element(nums)
        if len(nums) == 1:
            return 0

        for i in range(len(nums)):
            if i == 0 and nums[i] > nums[i + 1]:
                return i

            if i == len(nums) - 1 and nums[i] > nums[i - 1]:
                return i

            if (i > 0 and i < len(nums) - 1 and nums[i] > nums[i - 1] and nums[i] > nums[i + 1]):
                return i


        return 0
